src/praisonai-agents/praisonaiagents/tools/registry.py

```
@@ -36,6 +37,47 @@ def _get_entry_points():
+            # just use its get_schema() method and don't apply override again
+            return self.tool.get_schema()
+        else:
+            # For plain functions, generate schema and apply override if present
+            from .decorator import get_tool_schema
+            base_schema = get_tool_schema(self.tool)
+            
+            # Apply dynamic override if present
+            if self.schema_override is not None:
+                try:
+                    return self.schema_override(base_schema)
+                except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)